3.assess_generalizability/nbconverted/roc_qc_importance_eval.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Perform bootstrapping ROC AUC method to determine if performing QC within our pipelines is important and improves performance 
+# 
+# In this method, we have trained two models; one with QC'd data and the other without QC'd data (more noise). 
+# We apply the models to their respective holdout (`Plate_6`) dataset (e.g., QC'd model applied to QC'd data and no-QC model applied to no QC dataset).
+# We use bootstrapping, a method that repeatedly samples the dataset with replacement to create random subsets of the same size, where some cells might be duplicated or excluded, simulating variations in the population.
+# We calculate the ROC AUC for each subsample and plot as a histogram.
+# 
+# Our goal is to evaluate if QC is important enough to perform within our workflows where we see a higher performance in classification than if we performed no QC at all.
+
+# In[1]:
+
+
+import pathlib
+import joblib
+from joblib import load
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_auc_score
+from sklearn.utils import resample
+import matplotlib.colors as mcolors
+from scipy.stats import ttest_ind
+
+import sys
+
+sys.path.append("../utils")
+# from training_utils import get_X_y_data
+from roc_eval_utils import bootstrap_roc_auc
+
+
+# In[2]:
+
+
+figure_path = pathlib.Path("./figures")
+# make directory if it doesn't already exist
+figure_path.mkdir(exist_ok=True)
+
+
+# In[3]:
+
+
+# load in label encoder
+le = load(
+    pathlib.Path("../1.train_models/data/trained_nf1_model_label_encoder_qc.joblib")
+)
+
+# Print label mapping
+label_mapping = {label: le.transform([label])[0] for label in le.classes_}
+print(label_mapping)
+
+
+# In[4]:
+
+
+# Load the trained model
+no_QC_model = joblib.load(
+    pathlib.Path("../1.train_models/data/trained_nf1_model.joblib")
+)
+
+# Load the feature-selected QC plate 4 (ensure it includes both features and labels)
+plate_6_no_QC = pd.read_parquet(
+    pathlib.Path(
+        "/media/18tbdrive/1.Github_Repositories/nf1_schwann_cell_painting_data/3.processing_features/data/single_cell_profiles/Plate_6_sc_normalized.parquet"
+    )
+)
+
+# Drop rows where Metadata_genotype is HET
```

Comment:
might be helpful to describe why we do this

Reply:
Added!